### PR TITLE
#8220: Inject sandbox into document root instead of body

### PR DESCRIPTION
**Pull request by mnholtz**
## What does this PR do?

- Part of #8220
- Injects the sandbox iframe into the document root instead of the document body
- This is to address the hypothesis that web pages are removing the iframe from the DOM before the mod has the chance to use it (the assumption is that it is less likely for webpages to wipe things in the document root vs. the body)

## Discussion/Demo

- I'm able to consistently confirm that in the case of Docusign, the sandbox is always removed from the page after a certain point, causing a race with the mod component runtime: https://www.loom.com/share/e9ec1521b66e46fab93457a6ae391699
- This change resolves the issue in the case of Docusign, at least

## Future Work

- The next PR I'm preparing 
    - will remove [injectIframe memoization](https://github.com/pixiebrix/pixiebrix-extension/blob/7a6e3f620e10f1e3181a503cce95e82a9627eb36/src/sandbox/messenger/api.ts#L26-L26) so that we can ensure the iframe can be re-added on future brick runs if it does happen to get removed
    - and either leverage the `MutationObserver` approach I used to confirm behavior in the repro video (see above) or add retries to more thoroughly ensure that the iframe is on the page before executing a brick
    - throw more specific error to report if the iframe is not present on the page

## Checklist

- [ ] Added jest or playwright tests and/or storybook stories - It's possible to make a page on our playground that can replicate this scenario, but I'm not 100% if we want to write an e2e test for this before we know that this is the cause of the majority of user issues

For more information on our expectations for the PR process, see the
[code review principles doc](https://www.notion.so/pixiebrix/Code-Review-Principles-1ce7276b82a84d2a995d55ad85e1310d?pvs=4)

### CODE REVIEW

**Comment by twschiller** [comment]
@fregante any thoughts on using documentElement vs. body for the sandbox parent?

File: src/utils/injectIframe.tsx [hunk 1]
```diff
@@ -47,8 +47,8 @@ async function _injectIframe(
   Object.assign(iframe.style, style);
 
   // The body might not be available yet
-  await waitForBody();
-  document.body.append(shadowWrap(iframe));
+  await waitForDocumentRoot();
+  document.documentElement.append(shadowWrap(iframe));
```
Comment (twschiller) [inline src/utils/injectIframe.tsx:51]: Add comment explaining where we're adding to the documentElement

**Review by fregante — APPROVED** [review]
LGTM. We do the same for the sidebar and other widgets

File: src/utils/injectIframe.tsx (hunk 1 above)
Comment (mnholtz) [reply src/utils/injectIframe.tsx:53]: Thanks for catching that

**Pull request merged**